src/amqpvalue.c

```diff
@@ -1069,8 +1069,18 @@ AMQP_VALUE amqpvalue_create_string(const char* value)
     else
```

Comment:
Another place where there could be overflow as well:

In this block it should be checked the way you implemented in your PR.
The block assumes size_t == uint32_t, which is incorrect.
https://github.com/Azure/azure-uamqp-c/blob/28f603aaa0fa36b26e0eb4420dea1a9ef1cb3719/src/amqpvalue.c#L1177

There were no other locations I found that could have a type overflow followed by malloc.

Reply:
should I change the **IF** to

`if (length > UINT32_MAX)`